### PR TITLE
Bedrock Cross-Region support

### PR DESCRIPTION
**Pull request by csotomon**
# Description

The latest versions of Claude and Llama models now support Cross-Region inferences, which require adding the region prefix to the model identifier. This fix enables the use of this feature by adapting our model handling to include the regional prefix.

Fixes # (issue)
n/a
## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [ x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ x] Yes
- [ ] No

## Type of Change

- [ x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [ x] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py [hunk 1]
```diff
@@ -85,6 +109,18 @@
...
+    "us.anthropic.claude-3-sonnet-20240229-v1:0",
+    "us.anthropic.claude-3-5-haiku-20241022-v1:0",
+    "us.anthropic.claude-3-5-sonnet-20240620-v1:0",
+    "us.anthropic.claude-3-5-sonnet-20241022-v2:0",
+    "eu.anthropic.claude-3-haiku-20240307-v1:0",
+    "eu.anthropic.claude-3-opus-20240229-v1:0",
+    "eu.anthropic.claude-3-sonnet-20240229-v1:0",
+    "eu.anthropic.claude-3-5-haiku-20241022-v1:0",
+    "eu.anthropic.claude-3-5-sonnet-20240620-v1:0",
+    "eu.anthropic.claude-3-5-sonnet-20241022-v2:0",
```
Comment (logan-markewich) [inline llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py:123]: omg 😓 There has to be a better way to manage these names lmao

**Review by logan-markewich — COMMENTED** [review]
Should we also update bedrock converse?

**Comment by csotomon** [comment]
> Should we also update bedrock converse?

I don't know what is bedrock converse, I checked the code in my project and now I can use the Claude 3.5 V2 without problem

File: llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py (hunk 1 above)
Comment (csotomon) [reply llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py:123]: I feel the same way! I can try refactoring the code to help with this, but this was the quickest way I found to make it work.

**Comment by csotomon** [comment]
@logan-markewich  can U approve this MR?

For next week I will make a code refactoring for simplify the library maintenance.

**Comment by logan-markewich** [comment]
@csotomon bedrock converse is amazons preferred API. We have class exactly for it (BedrockConverse) -- it's much easier to use/maintain imo.

Anyways, can add it there myself

**Comment by csotomon** [comment]
> @csotomon bedrock converse is amazons preferred API. We have class exactly for it (BedrockConverse) -- it's much easier to use/maintain imo.
> 
> Anyways, can add it there myself

Perfect, I can help with  the BedrockConverse package and I will do it this weekend, but can U accept this pull request to update this package, I am using it in my projects.

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**